From 9a127c05dd3b9fd0e4967c8d59d549706e89607f Mon Sep 17 00:00:00 2001
From: Guillem Jover <guillem@hadrons.org>
Date: Tue, 13 Jun 2023 01:27:03 +0200
Subject: [PATCH 2/2] build: Fail the build if we have undefined symbols
Forwarded: https://github.com/axboe/liburing/pull/881
Last-Update: 2023-06-13

Better to fail the build than potentially generating a library that will
fail at run time when it cannot find the undefined symbols it requires,
due to underlinking (for example in the nolibc case).

Signed-off-by: Guillem Jover <guillem@hadrons.org>
---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index f57348b..951c48f 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -17,7 +17,7 @@ override CFLAGS += -Wno-unused-parameter \
 	$(LIBURING_CFLAGS)
 SO_CFLAGS=-fPIC $(CFLAGS)
 L_CFLAGS=$(CFLAGS)
-LINK_FLAGS=
+LINK_FLAGS=-Wl,-z,defs
 LINK_FLAGS+=$(LDFLAGS)
 ENABLE_SHARED ?= 1
 
-- 
2.40.1

